packages/sdk/android/src/main/java/com/primerioreactnative/PrimerRNHeadlessUniversalCheckoutListener.kt

```diff
@@ -240,13 +248,37 @@ class PrimerRNHeadlessUniversalCheckoutListener : PrimerHeadlessUniversalCheckou
 
   override fun onCheckoutAdditionalInfoReceived(additionalInfo: PrimerCheckoutAdditionalInfo) {
     if (implementedRNCallbacks?.isOnCheckoutAdditionalInfoImplemented == true) {
-      if (additionalInfo is PromptPayCheckoutAdditionalInfo) {
-        sendEvent?.invoke(
-          PrimerHeadlessUniversalCheckoutEvent.ON_CHECKOUT_ADDITIONAL_INFO.eventName,
-          JSONObject(Json.encodeToString(additionalInfo.toCheckoutAdditionalInfoRN())).apply {
-            remove("type")
+      when (additionalInfo) {
+        is PromptPayCheckoutAdditionalInfo -> {
+            sendEvent?.invoke(
+                PrimerHeadlessUniversalCheckoutEvent.ON_CHECKOUT_ADDITIONAL_INFO.eventName,
+                JSONObject(json.encodeToString(additionalInfo.toCheckoutAdditionalInfoRN())).apply {
+                  remove("type")
+                }
+            )
+        }
+        is AchAdditionalInfo.ProvideActivityResultRegistry -> {
+          val activityResultRegistry = (reactContext.currentActivity as? ComponentActivity)?.activityResultRegistry
+          if (activityResultRegistry == null) {
+            sendError?.invoke(
+              ErrorTypeRN.NativeBridgeFailed
+                errorTo PrimerRNHeadlessUniversalCheckoutStripeAchUserDetailsComponent.UNSUPPORTED_ACTIVITY_ERROR
+            )
+          } else {
+            additionalInfo.provide(activityResultRegistry)
           }
-        )
+        }
+
+        is AchAdditionalInfo.DisplayMandate -> {
+            PrimerRNAchMandateManager.acceptMandate = additionalInfo.onAcceptMandate
+            PrimerRNAchMandateManager.declineMandate = additionalInfo.onDeclineMandate
+            sendEvent?.invoke(
+                PrimerHeadlessUniversalCheckoutEvent.ON_CHECKOUT_ADDITIONAL_INFO.eventName,
+                JSONObject(json.encodeToString(additionalInfo.toCheckoutAdditionalInfoRN())).apply {
+                  remove("type")
```

Comment:
[nit]: could probably make an extension to remove any field or array of them

Reply:
Handled by creating removeType, as discussed privately.